resolve.js

```diff
@@ -0,0 +1,123 @@
+const assert = require('assert')
+const encoding = require('dat-encoding')
+const https = require('https')
+const NodeCache = require( "node-cache" )
```

Comment:
This is just an in-memory cache, right? Nothing persisted to disk?

What's the use case for an in-memory cache? When I join a cabal, my cabal-core instance will resolve a cabal URL exactly once, right?

Reply:
(Once we start building out cabal-client, I think that will be a great place to maintain an on-disk cache of resolved cabal URLs.)

Reply:
> This is just an in-memory cache, right? Nothing persisted to disk?
> 
> What's the use case for an in-memory cache? When I join a cabal, my cabal-core instance will resolve a cabal URL exactly once, right?

Yes. This is supposed to just be in-memory but I'll double check to be sure.

Do you think we should just take the caching out until we know how often it's used?

Reply:
+1 on not adding in-memory caching until there's a need for it. This also removes the need for a `close` API.